keyboards/mothwing/rev1/rev1.h

```diff
@@ -0,0 +1,22 @@
+// Copyright 2023 Tetsuro Tan (@tan-t)
+// SPDX-License-Identifier: GPL-2.0-or-later
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT( \
```

Comment:
Matrix positions should be specified in info.json, and this file can be deleted.

Reply:
hi @fauxpark .  
thank you for the review!  
I tried your suggestion locally, but I got this error:  
`#pragma error("<keyboard>.h is only optional for data driven keyboards - kb.h == bad
 times")`  

I think that there is some kind of option or flag to ensure that the keyboard is "data-driven", which means `rules.mk` and `config.h` are combined into `info.json`.   
and I see that the "data-driven" approach is somewhat (relatively) new and your suggestions are to follow the new convention, and I'm totally willing to do so.  
can you please show me how to turn on the "data-driven" mode for the keyboard?
thank you.

Reply:
btw, the commit is here: https://github.com/srac-technologies/qmk_firmware/commit/963923b2cf53a5f494eecc81fc48d66d32ca00a4

Reply:
Your layout in info.json is missing matrix positions for each key. If you do that, the `LAYOUT` macro in this .h file is no longer needed as QMK can generate it automatically at compile time.

Reply:
Thanks alot for the detailed advise.  
I tried a one but got another error that says `Matrix data out of bounds for layout LAYOUT at index 6 (R00): 0, 6`.  
I doubt that it is because the `matrix_pins` option has only 6 items for the columns.  
My PCB is for a split keyboard which has 6 columns per hand.

I tried the `qmk new_keyboard` command and chosed the `split_3x6_3` option (which seems most alike setup) to look into the generated `info.json` to copy or something, but the generated `info.json` was like 

```json
    "matrix_pins": {
        "cols": ["C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2", "C2"],
        "rows": ["D1", "D1", "D1", "D1"]
    },
```  

so I see there should be as many column pin items in `matrix_pins` as required matrix columns in `layout`.  

but my setup is 1 MCU per one hand, hence the matrix pins seems to be only 6 not 12.  
what should I do? maybe repeat the same pins two times?  

If possible, if you give me a good example of the new data-driven keyboard directory, it would be of great help.
thanks!

Reply:
Please read the split docs, it explains how you need to configure the matrix pins and layout.

Reply:
thank you kindly.  I finally got it built & it worked fine.